qualtran/bloqs/block_encoding.py

```diff
@@ -270,11 +362,11 @@ class ChebyshevPolynomial(Bloq):
         order: order of Chebychev polynomial.
 
     References:
-        [Quantum computing enhanced computational catalysis](
-            https://arxiv.org/abs/2007.14460). Page 45; Theorem 1.
+        [Quantum computing enhanced computational catalysis](https://arxiv.org/abs/2007.14460).
+            Page 45; Theorem 1.
     """
 
-    block_encoding: BlackBoxBlockEncoding
+    block_encoding: BlockEncoding
```

Comment:
This currently assumes that `G` in `BlockEncoding` is identity but there's no check anywhere. It's worth highlighting in the docstring at least that it's the case. 

Also, I think the standard way to construct chebyshev polynomial is to take the nth power of a (qubitization) walk operator.   

For example - 

1. Lemma 16 from https://arxiv.org/pdf/1511.02306
<img width="1079" alt="image" src="https://github.com/quantumlib/Qualtran/assets/7863287/3f52586b-2c00-4136-b84a-0df9ae2beb6c">

2. Also from https://quantum-journal.org/papers/q-2019-07-12-163/pdf/

<img width="1116" alt="image" src="https://github.com/quantumlib/Qualtran/assets/7863287/3f52aa86-8c48-4125-8d30-a5edee20f604">

 
 I belive Guang Hao uses this construction of alternating between B and B^{\daggar} to save one reflection instead of directly using W^{n} and optimize constant factors? 

Either way, it'll be nice to track this in an issue so we can extend the bloq to support arbitrary block encodings in a follow-up PR.

Reply:
Right, I would need to add a Identity PrepareOracle which I think I'll do in a follow up. I'll add a note / issue.

Reply:
Added #984

Reply:
Wait, it doesn't assume anything about $G$ in the block encoding, the reference provides an implementation which just has B (2|0><0|-1) B^, i.e. always reflection about zero no?

Reply:
it assumes that H is block encoded in the top-left corner; i.e `G` is identity ?

<img width="848" alt="image" src="https://github.com/quantumlib/Qualtran/assets/7863287/6069047f-b2b8-47f3-8514-00af11695637">

Reply:
You're right here's the earlier definition: 
![Uploading Screenshot 2024-05-21 at 4.20.50 PM.png…]()

Reply:
Quick question: since the first kind chebyshev polynomials are just QSP/Qubitization procedures with all the phase angles set to 0 why not just reuse that? I would think that would make a nicer decomposition.

Reply:
Probably didn't exist in the library at the time. Can you open an issue or add a comment to #984